flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/utils/OperationConverterUtils.java

```diff
@@ -106,13 +117,17 @@ public static Operation convertAddReplaceColumns(
         Map<String, String> newProperties = new HashMap<>(catalogTable.getOptions());
         newProperties.putAll(extractProperties(addReplaceColumns.getProperties()));
 
-        return new AlterTableSchemaOperation(
-                tableIdentifier,
+        CatalogTableImpl newTable =
                 new CatalogTableImpl(
                         builder.build(),
                         catalogTable.getPartitionKeys(),
                         newProperties,
-                        catalogTable.getComment()));
+                        catalogTable.getComment());
+        if (addReplaceColumns.isReplace()) {
```

Comment:
Because `ALTER TABLE REPLACE` syntax may also remove all columns before. 

> For example, "ALTER TABLE test_change REPLACE COLUMNS (a int, b int);" will remove column 'c' from test_change's schema.[1]

So I think we can support this after introducing DROP change. 

[1] https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-Add/ReplaceColumns